aesara/tensor/blas.py

```diff
@@ -630,8 +629,10 @@ def c_header_dirs(self, **kwargs):
         {PyErr_SetString(PyExc_NotImplementedError, "type(b) is not double or float"); %(fail)s;}
         """
 
+    # broadcast_xyz = None
+
     check_dims = """
-        if (Nx[0] != Nz[0])
+        if (Nx[0] !=1 && Nz[0] != 1 && Nx[0] != Nz[0])
```

Comment:
Only alter this for Gemm Op, no idea if other Gemm-related Ops are doing the right thing!